src/features/hotspots/settings/discovery/DiscoveryModeRoot.tsx

```diff
@@ -0,0 +1,197 @@
+  const [time, setTime] = useState(0)
+  const [hotspot, setHotspot] = useState(propsHotspot)
+  const { enableBack } = useHotspotSettingsContext()
+  const { result: userAddress } = useAsync(getSecureItem, ['address'])
+  const dispatch = useAppDispatch()
+  const requestInterval = useRef<NodeJS.Timeout>()
+  const clockInterval = useRef<NodeJS.Timeout>()
+  const recentDiscoveryInfo = useSelector(
+    (state: RootState) => state.discovery.recentDiscoveryInfo,
+  )
+  const infoLoading = useSelector(
+    (state: RootState) => state.discovery.infoLoading,
+  )
+  const selectedRequest = useSelector(
+    (state: RootState) => state.discovery.selectedRequest,
+  )
+  const requestId = useSelector((state: RootState) => state.discovery.requestId)
+
+  const fetchRecent = useCallback(() => {
+    if (!hotspot.address || !userAddress) return
+
+    dispatch(fetchRecentDiscoveries({ hotspotAddress: hotspot.address }))
+  }, [dispatch, hotspot.address, userAddress])
+
+  useEffect(() => {
+    if (viewState === 'begin') {
+      fetchRecent()
+    }
+  }, [fetchRecent, viewState])
+
+  useEffect(() => {
+    // TODO: Remove this and just use the hotspot from props
+
+    const lat = Number(Config.DISCO_HOTSPOT_LAT)
+    const lng = Number(Config.DISCO_HOTSPOT_LNG)
+    const address = Config.DISCO_HOTSPOT_ADDRESS
+    const name = Config.DISCO_HOTSPOT_NAME
+    if (lat && lng && address && name) {
+      const cloned = cloneDeep(propsHotspot)
+      cloned.lat = lat
+      cloned.lng = lng
+      cloned.address = address
+      cloned.name = name
+      setHotspot(cloned)
+    }
+  }, [propsHotspot])
+
+  const requestSeconds = useMemo(() => {
+    const insertDate = selectedRequest
+      ? new Date(selectedRequest.insertedAt)
+      : null
+    if (!insertDate) return 0
+
+    return getUnixTime(insertDate)
+  }, [selectedRequest])
+
+  const shouldPoll = useMemo(() => {
+    if (viewState === 'begin') return false
+
+    if (!selectedRequest) {
+      return true
+    }
+    return time - requestSeconds < DISCOVERY_DURATION_MINUTES * 60
+  }, [requestSeconds, selectedRequest, time, viewState])
+
+  useEffect(() => {
+    if (clockInterval.current) {
+      clearInterval(clockInterval.current)
+      clockInterval.current = undefined
+    }
+
+    if (!recentDiscoveryInfo?.serverDate || !shouldPoll) return
+
+    const serverTime = getUnixTime(new Date(recentDiscoveryInfo?.serverDate))
+    setTime(serverTime)
+    clockInterval.current = setInterval(() => {
+      setTime((val) => val + 1)
+    }, 1000)
+  }, [recentDiscoveryInfo?.serverDate, shouldPoll])
+
+  useEffect(() => {
+    if (!shouldPoll) {
+      if (requestInterval.current) {
+        clearTimeout(requestInterval.current)
+      }
+      return
+    }
+
+    const reqId = requestId || selectedRequest?.id
+    if (!reqId) return
+    dispatch(fetchDiscoveryById({ requestId: reqId }))
+
+    requestInterval.current = setInterval(() => {
```

Comment:
One thing to note here is this would stop if you put the app in the background.

We may want to use something like https://github.com/ocetnik/react-native-background-timer if we want it to keep running even if they minimize the app.